src/components/interaction-designer/block-editors/GroupMembershipEditor.vue

```diff
@@ -1,153 +1,188 @@
-            class="custom-control-input">
-          <label
-            class="custom-control-label font-weight-normal"
-            for="setFromExpression">
-            {{ 'flow-builder.set-group-membership-from-expression' | trans }}
+            :for="action.id">
+            {{ trans(action.localizationKey) }}
           </label>
         </div>
       </div>
 
-      <validation-message
-        #input-control="{ isValid }"
-        :message-key="`block/${block.uuid}/config/group_key`">
-        <div class="block-group-key">
-          <text-editor
-            v-model="groupKey"
-            :label="'flow-builder.group-label' | trans"
-            :label-class="'font-weight-bold'"
-            :placeholder="'flow-builder.enter-group-label' | trans"
-            :valid-state="isValid" />
-        </div>
-      </validation-message>
-
-      <validation-message
-        v-if="membershipAction === MEMBERSHIP_ACTION.SET_FROM_EXPRESSION"
-        #input-control="{ isValid }"
-        :message-key="`block/${block.uuid}/config/is_member`">
-        <expression-input
-          :label="'flow-builder.value-expression' | trans"
-          :placeholder="'flow-builder.enter-expression' | trans"
-          :label-class="'font-weight-bold'"
-          :current-expression="isMember"
-          :valid-state="isValid"
-          @commitExpressionChange="updateIsMemberExpression" />
-      </validation-message>
+      <vue-multiselect
+        v-if="isGroupListVisible"
+        v-model="selectedGroups"
+        :internal-search="!hasCustomSearch"
+        :is-loading="groupsLoading"
+        :options="groupOptions"
+        :taggable="!hasCustomSearch"
+        :multiple="true"
+        track-by="group_key"
+        label="group_name"
+        @seach-change="onSearchChange"
+        @tag="onGroupAdd" />
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import {IBlock} from '@floip/flow-runner'
+import {IBlock, ISetGroupMembershipBlockConfig, IGroupMembership} from '@floip/flow-runner'
 import {Component, Prop} from 'vue-property-decorator'
 import {namespace} from 'vuex-class'
+import VueMultiselect from 'vue-multiselect'
 import Lang from '@/lib/filters/lang'
-import {get} from 'lodash'
 import {mixins} from 'vue-class-component'
 
 const flowVuexNamespace = namespace('flow')
 
-const NULL_STRING_EXPRESSION = '@(null)'
-const TRUTHY_STRING_EXPRESSION = '@(true)'
-const EMPTY_STRING_EXPRESSION = ''
+type MembershipAction = {
+  id: string,
+  localizationKey: string,
+  value: string,
+}
+
+enum MEMBERSHIP_ACTION {
+  ADD = 'add',
+  REMOVE = 'remove',
+  CLEAR = 'clear',
+}
 
-@Component({})
+@Component({
+  components: {VueMultiselect},
+})
 export class GroupMembershipEditor extends mixins(Lang) {
   @Prop() readonly block!: IBlock
+  @Prop() readonly availableGroups?: IGroupMembership[]
+  @Prop({type: Boolean, default: false}) readonly groupsLoading!: boolean
+
+  customGroupOptions: IGroupMembership[] = []
```

Comment:
In the community version a user is allowed to tag groups, while in the Voto5 they to have to deal with a pre-defined list. `availableGropus` are pre-defined, and `customGroupOptions` are created by user.

Reply:
Ah ok, how about renaming it to `newCreatedGroups` or `newAddedGroups` ?

- [ ] rename it, or add a comment for context saying this will be used for tagged UI (without a pre-defined list)